py/sentry_streams/flink/flink_adapter.py

```diff
@@ -28,6 +29,27 @@ def __init__(self, config: MutableMapping[str, Any], env: StreamExecutionEnviron
         self.environment_config = config
         self.env = env
 
+    def assert_return_type(self, function: Callable[[Any], bool], expected_type: type) -> None:
+        return_type = get_annotations(function)["return"]
+        assert (
+            return_type is expected_type
+        ), f"Filter function {function.__name__} must return a bool, got {return_type} instead"
+
+    def get_function_from_step(self, step: Step) -> Any:
```

Comment:
Once we use function references instead of import paths in Map / Filter, we shouldn't need this logic around anymore